OsmAnd/src/net/osmand/plus/views/MapControlsLayer.java

```diff
@@ -350,6 +350,10 @@ protected void clickRouteWaypoints() {
 		}
 	}
 
+	public void stopNavigation() {
+		clickRouteCancel();
```

Comment:
Extract clickRouteCancel() to stopNavigation() and call stopNavigation() from clickRouteCancel()